### ISSUE TITLE
Loading themes

### ISSUE THREAD

**Issue by cope** [josdejong/jsoneditor#359]
In the [API Reference ](https://github.com/josdejong/jsoneditor/blob/master/docs/api.md)it says: 

> if you specify another one you need to make sure it is loaded

But it does not say **how** to load them? :(

I tried to add ace as a bower component, then to load both the js and the css theme file, but that did not work. I got a bunch of errors, so I am guessing that is not the way to load the theme.

Am I missing something?

**Comment by josdejong** [comment]
The docs of Ace explain how you can set a custom theme in Ace:

https://ace.c9.io/#nav=howto

Does that help? 

Do you use the minimalist version of of jsoneditor? This version has Ace excluded so you cannot run into any conflicts of two different versions of Ace being loaded at the same time.

**Comment by cope** [comment]
We are using the standard bower jsoneditor package: `"jsoneditor": "^5.5.10",`, and not minimalist, but full version.

The docs of Ace suggest only `editor.setTheme("ace/theme/twilight");` and claim that the themes are loaded on demand, which is opposite of "make sure it is loaded" :(

I know I'm doing _something_ wrong, I just can't figure out what exactly...

No matter what I do, I can only use these two themes:
'ace/theme/jsoneditor'
'ace/theme/textmate'
but no other :(

**Comment by cope** [comment]
Based on [this comment](https://github.com/josdejong/jsoneditor/issues/360#issuecomment-277521662), I tried `this.editor.aceEditor.setTheme("ace/theme/twilight");`, but this gives an error: 'GET http://localhost:3000/theme-twilight.js '

**Comment by josdejong** [comment]
You're indeed loading the theme correctly with `editor.setTheme("ace/theme/twilight")`. Ace indeed loads additional themes and plugins dynamically. That only works when the path to the folder of Ace (where all these files are located) is correct. You get a request error ` http://localhost:3000/theme-twilight.js`, which suggests that Ace doesn't have the correct path anymore, probably due to bundling of js code.

You can solve this in two ways:
- Fix the path of Ace source in your bundling tools. That may be hard.
- Load the theme yourself beforehand and bundling it in your app so Ace will not try to load it from a non-existing path. Simply add a `require('./correct/path/to/ace/theme/twilight')` on top of the file where you create Ace and JSONEditor.

**Comment by cope** [comment]
Is ace fully included in jsoneditor, or should I bower install ace as well?

This is what I find in the jsoneditor bowr folder:
![architect - c__git_architect - _client_classic_src_component_form_field_jso_2017-02-06_13-45-26](https://cloud.githubusercontent.com/assets/112515/22647581/ac475740-ec72-11e6-957c-9ecc28efe3e0.png)

**Comment by josdejong** [comment]
In the [`dist`](https://github.com/josdejong/jsoneditor/tree/master/dist) folder you will find two bundled versions of JSONEditor: the full version has Ace embedded, the minimalist version hasn't. The embedded version does not include additional plugins and themes of Ace, only the used ones by JSONEditor. So if you want to use additional plugins/themes you will have to load&bundle them yourself in your application.

If you already use Ace in other parts of your application, it would be a waste to bundle it twice and having two versions of Ace can give conflicts, so in that case you can best use the minimalist version of JSONEditor (which has Ace excluded).

**Comment by cope** [comment]
I see. Thanks.

TBH, with the other issues being resolved, I'm now using `ace/theme/textmate` for disabled and `ace/theme/jsoneditor` for enabled and this looks different enough to indicate behavior, so I will just stick with that.

Thanks for your help. :)

**Comment by josdejong** [comment]
:+1: you're welcome

**Comment by cope** [comment]
Just one thing I didn't get. How can I _load&bundle additional themes myself in my application_?

**Comment by josdejong** [comment]
If you create a bundle of your own application (for example with webpack) you can bundle anything in it. Typically you can include a theme from ace by requiring it somewhere in your app, just like JSONEditor itself bundles `mode/json` and `ext/searchbox` extensions of Ace here:

https://github.com/josdejong/jsoneditor/blob/master/src/js/ace/index.js#L11-L13

**Comment by cope** [comment]
Thanks. I'll try that.

**Comment by andrewbaldock** [comment]
Any luck @cope ?

**Comment by cope** [comment]
Sorry, no. I eventually gave up due to other issues pilling up :(

**Comment by josdejong** [comment]
@andrewbaldock is the issue you have about getting a custom version of Ace loaded, or about loading a custom theme for Ace?

**Comment by hoogw** [comment]
Can i change theme within api, like option: { theme: twitlighjt }

**Comment by josdejong** [comment]
You can select a theme when creating an editor, see docs. There is no API to change this once created, but you can probably do so via `myJsonEditor.aceEditor.setTheme(theme)`.

**Comment by acidjazz** [comment]
@josdejong i'm getting undefined when i 
```js
      this.editor = new window.JSONEditor(this.container, this.options)
      this.editor.set(this.presentation)
      console.log(this.editor.aceEditor)
```

**Comment by yzhaopost** [comment]
@josdejong I am experiencing the issue about loading different themes for ACE in JSONEditor. I am using the standard JSONEditor npm package: ` "jsoneditor": "^7.0.4",`. I can not set ACE editor theme which ACE provides (for example, tomorrow_night_bright).

Here is what I tried:
1. npm install brace so that I can import different ACE themes, for example
`import "brace/theme/tomorrow_night_bright";`
2. I tried to set this new theme using JSONEditor options before initialization. It does not work. But, there is no error.
![image](https://user-images.githubusercontent.com/45929614/66589747-0ffc3080-eb44-11e9-87fb-ac02e5f3b73d.png)
3. then, I tried to set the theme as you mentioned above (after the initialization). I got an error message:
![image](https://user-images.githubusercontent.com/45929614/66589962-8d27a580-eb44-11e9-81af-bd5fd77a3a79.png)
![image](https://user-images.githubusercontent.com/45929614/66590075-ce1fba00-eb44-11e9-839a-de8d48bc08cf.png)
Any help would be appreciated!
